src/reactionsystem_structure.jl

```diff
@@ -0,0 +1,1438 @@
+                             reduce(vcat, MT.namespace_equations.(systems, (ivs,));
+                                    init = Any[])]
+        return sort!(eqs; by = eqsortby)
+    end
+    return eqs
+end
+
+function MT.unknowns(sys::ReactionSystem)
+    sts = get_unknowns(sys)
+    systems = get_systems(sys)
+    if !isempty(systems)
+        sts = unique!([sts; reduce(vcat, namespace_variables.(systems))])
+        sort!(sts; by = !isspecies)
+        return sts
+    end
+    return sts
+end
+
+
+### Basic `ReactionSystem`-specific Accessors ###
+
+"""
+    get_species(sys::ReactionSystem)
+
+Return the current dependent variables that represent species in `sys` (toplevel system
+only).
+"""
+get_species(sys::ReactionSystem) = getfield(sys, :species)
+has_species(sys::ReactionSystem) = isdefined(sys, :species)
+
+"""
+    get_rxs(sys::ReactionSystem)
+
+Return the system's `Reaction` vector (toplevel system only).
+"""
+get_rxs(sys::ReactionSystem) = getfield(sys, :rxs)
+has_rxs(sys::ReactionSystem) = isdefined(sys, :rxs)
+
+"""
+    get_sivs(sys::ReactionSystem)
+
+Return the current spatial ivs, if the system is non-spatial returns an empty vector.
+"""
+get_sivs(sys::ReactionSystem) = getfield(sys, :sivs)
+has_sivs(sys::ReactionSystem) = isdefined(sys, :sivs)
+
+"""
+    get_networkproperties(sys::ReactionSystem)
+
+Return the current network properties of `sys`.
+"""
+get_networkproperties(sys::ReactionSystem) = getfield(sys, :networkproperties)
+
+"""
+    get_combinatoric_ratelaws(sys::ReactionSystem)
+
+Returns true if the default for the system is to rescale ratelaws, see
+https://docs.sciml.ai/Catalyst/stable/introduction_to_catalyst/introduction_to_catalyst/#Reaction-rate-laws-used-in-simulations
+for details. Can be overriden via passing `combinatoric_ratelaws` to `convert` or the
+`*Problem` functions.
+"""
+get_combinatoric_ratelaws(sys::ReactionSystem) = getfield(sys, :combinatoric_ratelaws)
+
+"""
+    combinatoric_ratelaws(sys::ReactionSystem)
+
+Returns the effective (default) `combinatoric_ratelaw` value for a compositional system,
+calculated by taking the logical or of each component `ReactionSystem`. Can be overriden
+during calls to `convert` of problem constructors.
+"""
+function combinatoric_ratelaws(sys::ReactionSystem)
+    crl = get_combinatoric_ratelaws(sys)
+    subsys = Iterators.filter(s -> s isa ReactionSystem, get_systems(sys))
+    mapreduce(combinatoric_ratelaws, |, subsys; init = crl)
+end
+
+"""
+    species(network)
+
+Given a [`ReactionSystem`](@ref), return a vector of all species defined in the system and
+any subsystems that are of type `ReactionSystem`. To get the species and non-species
+variables in the system and all subsystems, including non-`ReactionSystem` subsystems, uses
+`unknowns(network)`.
+
+Notes:
+- If `ModelingToolkit.get_systems(network)` is non-empty will allocate.
+"""
+function species(network)
+    sts = get_species(network)
+    systems = filter_nonrxsys(network)
```

Comment:
where is `filter_nonrxsys`? it should precede this function.

Reply:
My impression was that this was meant as a general API function as well (although we technically do not export it I guess). I figured that it made sense to roughly sore the network API functions from standard to obscure, and those felt like it was at the far end. But if you want it internal I can move it up.

Reply:
I’d still prefer to stick with the convention of functions appearing before their use as I had previously.